### DISCUSSION TITLE
optim too slow on AWS EC2 instance

### DISCUSSION THREAD

**Discussion by raamana** [optuna/optuna#5725] [Q&A]
### Expected behavior

first of all, I love the library and **thank you** for open sourcing and maintaining it. 

Issue:
I am optimizing a forecasting model from Darts with Optuna and the individual trials finish in about 2 mins on M3 Max Macbook pro ( Number of Cores: 14 (10 performance and 4 efficiency)) for a wide variety of hyper param configurations. When I run the exact same thing on an EC2 instance (type `m8g.8xlarge` with 32 vCPU and 128 GB RAM), and they never finish even when given 30 mins per trial (accounting for CPU vs GPU differences). What's going on?

I am using a timeout decorator as suggested to prune long-running or hungup trials, and on EC2 instance, every single trial is getting pruned, even with a different predictive model. 

```python
@timeout_decorator.timeout(max_run_time_per_trial,  # in seconds
                           timeout_exception=optuna.TrialPruned,  # trial result will be noted as pruned
                           use_signals=True)  # needed for process-based parallelization with n_jobs=1
def objective(trial):
```

I know this might not be optuna-specific issue but I would appreciate any constructive feedback and pointers. 

### Environment

- Optuna version: 4.0.0 (both places)

- Local Python version: 3.12.4
- Local OS: macOS-14.6.1-arm64-arm-64bit

- EC2 instance OS: Linux-6.8.0-1016-aws-aarch64-with-glibc2.39
- EC2 Python version:3.12.3

### Error messages, stack traces, or logs

no errors or warnings - trials just keep running!

### Steps to reproduce

i am not allowed to share code or data but I can try to write down a simplest example to reproduce the issue that does not involve any corporate restrictions, but I would like your feedback on the above before I develop this example

**Comment by nzw0301** [comment]
I suggest taking some profiles to identify which part is slower on ec2 than your laptop.

**Reply by raamana** [reply]
Thanks Kento - Good idea! I was going to do that but I just wanted to if you or the other users might be aware of anything that can identify the issue.

**Reply by nzw0301** [reply]
I'm afraid that I don't aware of such an issue. sorry
